library/network_connections.py

```diff
@@ -837,6 +859,40 @@ def connection_create(self, connections, idx, connection_current=None):
                     NM.SETTING_WIRED_SPEED, connection["ethernet"]["speed"]
                 )
 
+            # Handle ethtool features
+            # If NM supports ethtool settings:
+            # Set them accordingly
+            # If NM does not support ethtool settings:
+            # (a) if some are specified, fail
+            # (b) if none are specified, ignore them
+            ethtool_features = connection["ethtool"]["features"]
+            try:
+                s_ethtool = NM.SettingEthtool.new()
+            except AttributeError:
+                # NM is too old to support SettingEthtool
+                s_ethtool = False
```

Comment:
Could you assign here `None`? `s_ethtool` should be a NMSetting instance and not a boolean. Note that like most `GObject` constructors (*), `NM.SettingEthtool.new()` never fails -- unless libnm is too old.

Depending on how you look at it it, `None` is what pygobject would make for a non-existing GObject. Hence, one could argue that `None` is still a sensible value for a `GObject`/`NMSetting`. But `False` is an entirely different type. It seems unnecessarily confusing to mix booleans and GObject value for the same variable.

Of course, there is no difference in practice.

---


(*) Only GObjects that implement `GAsyncInitable` or `GInitable` (which few do) can fail to initialize during `g_async_initable_init_async` or `g_initable_init()`, but even then it's always a step after `g_object_new()`. `g_object_new()` can never fail, what can fail is to initiate the object afterwards. `g_initiable_new()` is basically `g_object_new() + g_initable_init()` together, and that might fail again.

Reply:
ok, makes sense now.